packages/teuchos/comm/src/Teuchos_StackedTimer.cpp

```diff
@@ -56,6 +56,11 @@ StackedTimer::LevelTimer::findBaseTimer(const std::string &name) const {
 BaseTimer::TimeInfo
 StackedTimer::LevelTimer::findTimer(const std::string &name, bool& found) {
   BaseTimer::TimeInfo t;
+  auto full_name = get_full_name();
+  if (full_name.size() > name.size())
```

Comment:
I think this is fine in context of the `strncmp` call below.  On the other hand, now you're doing two string comparisons.  I would have written the code like this:
```c++
const std::string full_name = get_full_name ();
if (full_name != name) {
  return t;
}
t = BaseTimer::TimeInfo(this);
found = true;
```